File: 900/17_Array_Cloning_Technique.py
# https://codeforces.com/problemset/problem/1665/B
# Gives TLE For CF
import sys
import math

T = int(input())
for _ in range(T):
    n = int(sys.stdin.readline())
    a = list(map(int, sys.stdin.readline().split()))
    freq_map = {}
    max_freq = 0
    
    for num in a:
        freq_map[num] = freq_map.get(num, 0) + 1
        if freq_map[num] > max_freq:
            max_freq = freq_map[num]
    
    left_places=n-max_freq
    copy_ops=0 if left_places<=0 else math.ceil(math.log2(left_places/max_freq + 1))
    print(left_places + copy_ops)

File: 900/17_Array_Cloning_Technique.cpp
// https://codeforces.com/problemset/problem/1665/B
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll t;
    cin >> t;
    while(t--){
        ll n;
        cin >> n;
        vector<ll> a(n);
        for(ll i=0; i<n; ++i) cin >> a[i];
        map<ll, ll> mp;
        ll freq=0;
        for(ll i=0;i<n;++i){
            mp[a[i]]++;
            if(mp[a[i]]>freq){
                freq=max(mp[a[i]], freq);
            }
        }
        ll ops=0;
        ll left_places=n-freq;
        ll cur_copy=freq;
        while(left_places>0){
            ops+=1;
            ops+=min(left_places,cur_copy);
            left_places-=min(left_places,cur_copy);
            cur_copy<<=1;
        }
        cout << ops << endl;
    }
}